### PR TITLE
33 add profile page

### LINKED ISSUE #33: Add: profile page
Is your feature request related to a problem? Please describe.
N/A

Describe how the solution should behave
The profile page should display the user's current information. The edit modal should be also implemented in this PR.

Screenshots of the design/if any
image
image

Additional context
Formik for input validation.

### PR DESCRIPTION
**Pull request by aliSalehAbdulhadi**
1. Added userProfile page
2. Added userProfileSetting modal
3. Added latestQuestion and latestAnswers files to render for the active user
4. Refactored Sign in and Sign up code to a new file to switch between them and UserProfile files easily when a user log in
5. Added FileReader to Dropzone in addQuestion and userProfileSetting modals to be able to read data files
6. Removed prettyBytes library since snapshot test is not reading it
7. Fixed some visual bugs

### CODE REVIEW

File: src/components/Answer/Answer.jsx [hunk 1]
```diff
@@ -56,6 +56,10 @@ function Answer({ children, image, username }) {
           </div>
         </div>
         {children}
+        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Dolorem eos
+        tempore tenetur ea suscipit, quisquam adipisci, tempora sequi nostrum
+        omnis, autem labore eius accusamus quas voluptatibus? Iste odit itaque
+        cumque!
```
Comment (MuhannedNoman) [inline src/components/Answer/Answer.jsx]: Why is this text here? 
Isn't the children the actual question?

File: src/components/modals/AddQuestion/AddQuestion.jsx [hunk 2]
```diff
@@ -6,17 +6,18 @@ import * as Yup from 'yup';
...
 
 const SignInSchema = Yup.object().shape({
   title: Yup.string().min(3).max(24).required(),
   question: Yup.string().email().required(),
-  attachment: Yup.string().min(8).max(32).required(),
 });
 
 function AddQuestion({ open, setOpen }) {
+  // eslint-disable-next-line no-unused-vars
+  const [newPic, setNewPic] = useState('');
```
Comment (MuhannedNoman) [inline src/components/modals/AddQuestion/AddQuestion.jsx:22]: What is the unused variable?

File: src/components/modals/profileSetting/ProfileSetting.jsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+const ContactSchema = Yup.object().shape({
+  firstName: Yup.string().min(2).max(10),
+  lastName: Yup.string().min(2).max(10),
+  title: Yup.string().min(2).max(24),
+});
+function ProfileSetting({ open, setOpen }) {
+  // eslint-disable-next-line no-unused-vars
+  const [newProfilePic, setProfilePic] = useState('');
+  // eslint-disable-next-line no-unused-vars
+  const [newCoverPic, setCoverPic] = useState('');
```
Comment (MuhannedNoman) [inline src/components/modals/profileSetting/ProfileSetting.jsx:19]: Is the setter not being used? You can remove them btw.

File: src/components/navbar/Navbar.jsx [hunk 4]
```diff
@@ -13,18 +13,16 @@ import {
   QUESTIONS_ROUTE,
 } from '../../routes';
 import AddQuestion from '../modals/AddQuestion/AddQuestion';
-import SignUp from '../modals/SignUp/SignUp';
-import SignIn from '../modals/SignIn/SignIn';
+// import UserAuth from '../userSection/UserAuth';
```
Comment (MuhannedNoman) [inline src/components/navbar/Navbar.jsx:16]: Let's remove any unused importes.

File: src/components/profile/latestQuestions/LatestQuestions.jsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import Question from '../../Question/Question';
+
+function LatestQuestions() {
+  return (
+    <div>
+      <Question />
```
Comment (MuhannedNoman) [inline src/components/profile/latestQuestions/LatestQuestions.jsx:6]: Is this component not getting any data as props. Like questions or anything?

**Review by MuhannedNoman — CHANGES_REQUESTED** [review]
Hey @aliSalehAbdulhadi good work here, left you some comments to see what you think?

File: src/components/Answer/Answer.jsx (hunk 1 above)
Comment (aliSalehAbdulhadi) [reply src/components/Answer/Answer.jsx]: I forgot it there. I added it just to see the width of the answer component. I'll remove it

File: src/components/modals/AddQuestion/AddQuestion.jsx (hunk 2 above)
Comment (aliSalehAbdulhadi) [reply src/components/modals/AddQuestion/AddQuestion.jsx:22]: This is what we gonna use to store the pic locally before dispatching them with the form before dispatching the question. Let me know if you want me to remove them for now.

File: src/components/modals/profileSetting/ProfileSetting.jsx (hunk 3 above)
Comment (aliSalehAbdulhadi) [reply src/components/modals/profileSetting/ProfileSetting.jsx:19]: This is what we gonna use to store the pic locally before dispatching them with the form before updating our profile.
Let me know if you want me to remove them for now.

File: src/components/navbar/Navbar.jsx (hunk 4 above)
Comment (aliSalehAbdulhadi) [reply src/components/navbar/Navbar.jsx:16]: We will use this in a ternary operator to switch between the user name and pic section on the navbar if there is a user or keep the sign in and sign up section if there is no user

File: src/components/profile/latestQuestions/LatestQuestions.jsx (hunk 5 above)
Comment (aliSalehAbdulhadi) [reply src/components/profile/latestQuestions/LatestQuestions.jsx:6]: This component gets its data from the API when we fetch the incoming data.

**Review by MuhannedNoman — DISMISSED** [review]

**Review by MuhannedNoman — DISMISSED** [review]

**Review by MuhannedNoman — APPROVED** [review]

**Pull request merged**